metricbeat/helper/kubernetes/ktest/ktest.go

```diff
@@ -19,42 +19,59 @@ package ktest
 
 import (
 	"io/ioutil"
+	"os"
+	"path/filepath"
 	"strings"
 	"testing"
 
 	p "github.com/elastic/beats/v7/metricbeat/helper/prometheus"
 	"github.com/elastic/beats/v7/metricbeat/helper/prometheus/ptest"
 )
 
-// GetTestCases Build test cases from the files and returns them
-func GetTestCases(files []string) ptest.TestCases {
+func getFiles(folder string) ([]string, error) {
+	var files []string
```

Comment:
There is some nice [recommendation](https://stackoverflow.com/questions/59734706/how-to-resolve-consider-preallocating-prealloc-lint) on how to preallocate a slice with known maximum capacity

Reply:
But the size of that directory is not always the same. I could add a maximum size, but then we would have to be updating the number every time we bump the KSM version.

Reply:
Which directory you mean? The linter is complaining about the files slice initilisation. Its highest value will always be equal to the length of entries variable.